File: app/src/main/java/com/example/dailyupdate/viewmodels/GitHubViewModel.java
package com.example.dailyupdate.viewmodels;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.ViewModel;

import com.example.dailyupdate.data.models.GitHubRepo;
import com.example.dailyupdate.repositories.GitHubRepository;

import java.util.List;

public class GitHubViewModel extends ViewModel {

    private GitHubRepository mRepo;
//    private MutableLiveData<Boolean> mIsUpdating = new MutableLiveData<>();

    public GitHubViewModel() {
        mRepo = GitHubRepository.getInstance();
    }

    public LiveData<List<GitHubRepo>> getGitHubRepoList() {
        return mRepo.getGitHubRepoList();
    }

    public void searchGitHubRepoList(String searchKeyword, String sortBy){
        mRepo.searchRepositoriesFromGitHubApi(searchKeyword, sortBy);

    }

    public void searchGitHubRepoListWithOrder(String searchKeyword, String sortBy,
                                              String sortOrder){
        mRepo.searchRepositoriesFromGitHubApiWithOrder(searchKeyword, sortBy, sortOrder);
    }

//    public LiveData<Boolean> getIsUpdating(){
//        return mIsUpdating;
//    }
}

File: app/src/main/java/com/example/dailyupdate/viewmodels/GitHubViewModel.kt
package com.example.dailyupdate.viewmodels

import androidx.lifecycle.*
import com.example.dailyupdate.data.models.GitHubRepo
import com.example.dailyupdate.repositories.GitHubRepository
import com.example.dailyupdate.utilities.Resource

class GitHubViewModel : ViewModel() {
    private val mRepo: GitHubRepository

    init {
        mRepo = GitHubRepository.instance
    }

    private val _gitHubRepoList = MediatorLiveData<Resource<List<GitHubRepo>>>()
    val gitHubRepoList: LiveData<Resource<List<GitHubRepo>>>
        get() = _gitHubRepoList

    fun searchGitHubRepoList(searchKeyword: String,
                             sortBy: String,
                             sortOrder: String = "desc") {
        _gitHubRepoList.addSource(
                mRepo.searchRepositoriesFromGitHubApiWithOrder(
                        searchKeyword,
                        sortBy,
                        sortOrder,
                        viewModelScope),
                _gitHubRepoList::setValue)
    }
}